request.go

```diff
@@ -64,7 +70,24 @@ func (r *Request) Encode(v interface{}) {
 	r.Header.Set("Content-Type", "application/json")
 }
 
-// EncodeAsProtobuf serialises the passed object as protobuf into the body
+// EncodeAsProtoJSON serialises the passed object as ProtoJSON into the body (and sets appropriate headers).
+func (r *Request) EncodeAsProtoJSON(m proto.Message) {
```

Comment:
I wonder if we should expose this as a public function at all? I can't think of a use-case where we'd want to explicitly send out protojson delibrately. We need to continue to support in the Decode but I don't believe we should explicitly add a new method to Encode

If we do intend to keep this in the interface, can we add some explicit tests please so we don't have a further regression in future 🙏

Reply:
I've removed this endpoint and just left legacy JSON / protobuf